### PR TITLE
kn release v0.10.0

### PR DESCRIPTION
**Pull request by navidshaikh**
- Add ci-operator config and job for knative-client release-v0.10.0 branch
 - Mirror kn v0.10.0 image to quay

### CODE REVIEW

File: ci-operator/jobs/openshift/private-ci-testing/openshift-private-ci-testing-master-presubmits.yaml [hunk 1]
```diff
@@ -41,17 +41,17 @@ presubmits:
           requests:
             cpu: 10m
         volumeMounts:
-        - mountPath: /usr/local/github-ssh-credentials-openshift-bot
-          name: github-ssh-credentials-openshift-bot
+        - mountPath: /usr/local/github-credentials
+          name: github-credentials-openshift-ci-robot-private-git-cloner
```
Comment (rhuss) [inline ci-operator/jobs/openshift/private-ci-testing/openshift-private-ci-testing-master-presubmits.yaml]: so has there some change in the ay authentication is done in the ci operator ? (just out of curiosity)

Comment (navidshaikh) [reply ci-operator/jobs/openshift/private-ci-testing/openshift-private-ci-testing-master-presubmits.yaml]: Yup, these jobs configs are auto-generated using `registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest` container (locally) and we've to include part of this PR (there is check in this PR's CI which verifies the files diff).

**Comment by droslean** [comment]
@navidshaikh Can you please rebase?

File: ci-operator/jobs/openshift/private-ci-testing/openshift-private-ci-testing-master-presubmits.yaml (hunk 1 above)
Comment (navidshaikh) [reply ci-operator/jobs/openshift/private-ci-testing/openshift-private-ci-testing-master-presubmits.yaml]: Rebased, the changes are in master branch already.

**Pull request merged**